awstesting/assert_xml_go1.8.go

```diff
@@ -0,0 +1,35 @@
+// +build !go1.9
+
+package awstesting
+
+import (
+	"bytes"
+	"encoding/xml"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/private/protocol/xml/xmlutil"
+)
+
+// AssertXML verifies that the expect xml string matches the actual.
+// Elements in actual must match the order they appear in expect to match equally
+//
+// Deprecated: Update to Go 1.9 or higher.
+func AssertXML(t *testing.T, expect, actual string, msgAndArgs ...interface{}) bool {
+	buffer := bytes.NewReader([]byte(expect))
+	decoder := xml.NewDecoder(buffer)
+
+	expectVal, err := xmlutil.XMLToStruct(decoder, nil)
+	if err != nil {
+		t.Fatalf("failed to umarshal xml to struct: %v", err)
+	}
+
+	buffer = bytes.NewReader([]byte(actual))
+	decoder = xml.NewDecoder(buffer)
+
+	actualVal, err := xmlutil.XMLToStruct(decoder, nil)
+	if err != nil {
+		t.Fatalf("failed to umarshal xml to struct: %v", err)
```

Comment:
Nit: why is this Fatal but the oher go_1.9 version is not?

Reply:
Pushed update to this PR fixing all of this up and no longer depending on smithy-go due to go version support. Switched to copy/pasta with internal pkg.